### PR TITLE
Update release documentation with the latest improvements

### PR DESCRIPTION
**Pull request by sergiou87**
## Description

This PR updates the release docs with mentions to the changes from #14727 and #14761 

## Release notes

Notes: no-notes

### CODE REVIEW

File: docs/process/releasing-updates.md [hunk 1]
```diff
@@ -135,6 +132,8 @@ Add your new changelog entries to `changelog.json`, update the version in `app/p
 
 If a maintainer would like to backport a pull request to the next release, it is their responsibility to co-ordinate with the release owner and ensure they are fine with accepting this work.
 
+After pushing the branch, an action will create a release Pull Request for it. If that action fails for whatever reason, you can fall back to using the `yarn draft-release:pr` command, or create it manually.
```
Comment (tidy-dev) [inline docs/process/releasing-updates.md]:
```suggestion
After pushing the branch, a [GitHub Action](https://github.com/desktop/desktop/blob/development/.github/workflows/release-pr.yml) will create a release Pull Request for it. If that action fails for whatever reason, you can fall back to using the `yarn draft-release:pr` command, or create it manually.
```

**Review by tidy-dev — CHANGES_REQUESTED** [review]
Just a minor suggestion - If I was new to Github I may not understand what "an action" is referring to. I added a suggested change to link to the .yml so one could easily find/see how the pull request is created. :)

File: docs/process/releasing-updates.md (hunk 1 above)
Comment (sergiou87) [reply docs/process/releasing-updates.md]: Yes! Much better, thank you 🙇

**Review by tidy-dev — APPROVED** [review]
✨  Thanks for that change!

**Pull request merged**